### ISSUE TITLE
Malformed response for recent/failed jobs containing binary data - take 2

### ISSUE THREAD

**Issue by ragingdave** [laravel/horizon#1007]
- Horizon Version: 5.7.4
- Laravel Version: 8.35.1
- PHP Version: 7.4.3

### Description:
Bringing #628 back up based on some new issue responses found in the main laravel/framework repo.
The issue still remains of horizon breaking on recent/failed jobs with binary data in payload/exception. This happens for me at least, because the exception contains binary data due to using a binary uuid. If a job fails with that data found in the exception the failed jobs area breaks.

I bring this back up because firstly, it's still happening, and secondly because of a more recent response around json decode handling from https://github.com/laravel/framework/issues/33928. That issue brings up a similar issue around binary information, and the response there was that it was something that should be handled in user land. 

Specifically, "Laravel will not deal with Malformed UTF-8 characters. This is a decision for the user-land." 

I do agree partially with that, but in this case, horizon is basically where that handling would need to take place as it's effectively "user land" even though it's first party. I argue this because there's nothing in userland that could be modified to handle converting an exception during job handling into something that's json compatible without modification of the horizon decode logic found here: https://github.com/laravel/horizon/blob/5.x/src/Http/Controllers/FailedJobsController.php#L114

Basically there is nothing that a user can do in user land to avoid the binary data, as the user doesn't have access outside of the vendor directory to remove the binary data or modify the exception caught internally in horizon. This issue makes the failed job UI completely unusable and has since I submitted the original ticket.

See the original ticket for further description/discussion https://github.com/laravel/horizon/issues/628.

**Comment by driesvints** [comment]
Sorry @ragingdave I totally missed this one. I didn't get to this today. I'll try to take a look as soon as I'm able to.

**Comment by driesvints** [comment]
Hey @ragingdave, thanks for the thorough issue. That was very helpful. I'd like to help out here. Can you maybe give me an easy way to reproduce the issue? Maybe a very basic queued job that contains a binary uuid?

**Comment by ragingdave** [comment]
So in any app that has horizon, on any model that connects at least for my case to mysql (but that shouldn't really matter), add the following code to a route/command that you can dispatch. For example, a route:

```
use Illuminate\Support\Str;
use Ramsey\Uuid\Uuid;

Route::get('test', function () {
    dispatch(function () {
        $binary = Uuid::fromString(Str::uuid())->getBytes();
        $model = new Pick\Any\Eloquent\Model();
        $model->uuid_field = $binary;
        $model->save();
    });
});
```
This works by forcing a database level exception, in this case a missing field, to fail the job. You wind up with binary data that you can't control entering the failed jobs table, thus breaking the horizon failed jobs view.

**Comment by driesvints** [comment]
@ragingdave thanks. I can indeed reproduce it that way. But I have no idea unfortunately on how we can fix that. I'll discuss this with the team.

**Comment by driesvints** [comment]
Hey @ragingdave. We recognise the bug but can't spend time on this in the short term. If I ever have time on hand I'll try to dig into this again but for the moment appreciating help with this.

**Comment by taylorotwell** [comment]
I pushed a fix for it - haven't tagged yet though.

**Comment by driesvints** [comment]
Should be tagged on Tuesday. Thanks for reporting!
